storage/blob.go

```diff
@@ -627,3 +627,13 @@ func pathForResource(container, name string) string {
 	}
 	return fmt.Sprintf("/%s", container)
 }
+
+func (b *Blob) respondCreation(resp *storageResponse, bt BlobType) error {
+	readAndCloseBody(resp.body)
+	err := checkRespCode(resp.statusCode, []int{http.StatusCreated})
+	if err != nil {
+		return err
+	}
+	b.Properties.BlobType = bt
```

Comment:
Should this be set before returning if err != nil?

Reply:
I prefer not to modify the receiver if the operation was not successful

Reply:
Thinking about this more I agree with you, if the creation fails then no blob exists so it's ok to not set the type.